### PR TITLE
Export UCT - change of status

### PR DESCRIPTION
**Pull request by benrejebmoh**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [ ] The commit message follows our guidelines
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**Does this PR already have an issue describing the problem ?** *If so, link to this issue using `'#XXX'` and skip the rest*

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*

BugFix

**What is the current behavior?** *(You can also link to an open issue here)*
In the exported CGM, the status of all the tie-lines are changed from 0 = real element to 1 = equivalent element.
This is also a problem for all X-nodes, which are changed from real to equivalent node status
In also happens on the other way round: internal equivalent lines become real ones after the merging. 

**What is the new behavior (if this is a feature change)?**

Keeping Ucte element statut as property in import then use it in export.
For DanglingLines we keep both UcteNodeStatus and UcteElementStatus.
For XNode Creation in export if there is any incoherence between the two network, we use UcteNodeStatus.REAL as default value.
Example:
network1:
##ZXX
XXXXXX11 XNODE 1      0 0        3.96708 0.00000 0.00000 0.00000
network2:
##ZXX
XXXXXX11              1 0        3.96708 0.00000 0.00000 0.00000
Merged network will be:
##ZXX
XXXXXX11 XNODE 1      0 0        0.00000 0.00000 0.00000 0.00000

**Does this PR introduce a breaking change or deprecate an API?** *If yes, check the following:*
- [ ] The *Breaking Change* or *Deprecated* label has been added
- [ ] The migration guide has been updated in the github wiki *(What changes might users need to make in their application due to this PR?)*

**Other information**:

(if any of the questions/checkboxes don't apply, please delete them entirely)

### CODE REVIEW

**Comment by mathbagu** [comment]
@benrejebmoh Are you sure this cannot be managed using the fictitious status of IIDM. This flag is now available on all equipments, so you should be able to initialize the fictitious attribute from UCTE, and be able to export it back?

**Comment by benrejebmoh** [comment]
@mathbagu that could work for XNodes, but how i could manage UcteElementStatus otherwise?

**Comment by mathbagu** [comment]
> @mathbagu that could work for UcteNodeStatus, but how i could manage UcteElementStatus otherwise?

Exactly the same: I'm not sure the fictitious flag is correctly initialized in the import, but it should be possible to retrieve the status using it:
- In/Out means connected/disconnected for a branch, and close/open for a switch (coupler)
- Fictitious for real/equivalent equipment.

Do I miss something?

**Comment by benrejebmoh** [comment]
@mathbagu , 
Thank you for the explanation.
For the danglingLines i need to keep both node status and element status. that's why in createDanglingLine method you will find  addStatusProperty(ucteLine, dl) and addStatusProperty(xnode, dl).
In addition for tieLine creation i need to identify element status for each half (exactly like element name).
That's why i made the choice to work with properties for every thing related to status managment.
Indeed i can refactor my code to work with isFictious flag when it's possible and with properties otherwise.
What do you recommend?

**Comment by mathbagu** [comment]
I don't know which is the best way to implement this:
- using a couple of properties
- adding a fictitious flag on both HalfLine of a TieLine
- using the MergedXnode extension

@sebamurg: any suggestion?

**Comment by murgeyseb** [comment]
> I don't know which is the best way to implement this:
> 
>     * using a couple of properties
> 
>     * adding a fictitious flag on both HalfLine of a TieLine
> 
>     * using the MergedXnode extension
> 
> 
> @murgeyseb : any suggestion?

For consistency with elementName properties, I would prefer using properties for that, at least while we do not have dedicated extensions for storing UCTE dedicated data. I do not know who uses MergedXnode extension, because we do not use it in merging server, nor in FARAO.

**Comment by benrejebmoh** [comment]
So is it ok?

**Comment by murgeyseb** [comment]
After a short talk with Mathieu, it seems that it may be better to add a fictitious flag for half lines. I try to do a modification of the PR to take it into account

File: iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/TieLineAdapter.java [hunk 1]
```diff
@@ -34,4 +34,9 @@ public HalfLine getHalf1() {
     public HalfLine getHalf2() {
         return ((TieLine) getDelegate()).getHalf2();
     }
+
+    @Override
+    public HalfLine getHalf(Side side) {
+        return ((TieLine) getDelegate()).getHalf(side);
```
Comment (mathbagu) [inline iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/TieLineAdapter.java:40]: I wonder if we should override getDelegate() to return a TieLine and avoid casting everywhere

Comment (murgeyseb) [reply iidm/iidm-mergingview/src/main/java/com/powsybl/iidm/mergingview/TieLineAdapter.java:40]: As it is an AbstractAdapter<Line>, I do not think it is possible to change return type of getDelegate to TieLine.

**Pull request merged**